folium/plugins/time_slider_choropleth.py

```diff
@@ -156,3 +164,14 @@ def __init__(self, data, styledict, name=None, overlay=True, control=True,
 
         self.timestamps = timestamps
         self.styledict = styledict
+        if init_timestamp >= 0:
```

Comment:
These error messages are very nice. I'm thinking though, could you combine them into a single check? It's less verbose that way, but will do the same. Just check if the value is between the expected lower and higher values.